File: Models/Notification.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace MeetingPlanner.Models
{
    public class Notification
    {
        [Key]
        public Guid Id { get; set; } = Guid.NewGuid();

        [Required]
        public int Quantity { get; set; }
        
        [Required]
        public NotificationUnit Unit { get; set; }
    }
}

File: MeetingPlanner/ClientApp/src/app/shared/model/notification.ts
import { NotificationUnit } from './notification-unit';

export class Notification {
  id?: string;
  quantity?: number;
  unit?: NotificationUnit;

  constructor(obj) {
    for (const prop in obj) {
      if (this[prop]) {
        this[prop] = obj[prop];
      }
    }
  }
}
